src/test/spec/runner/mod.rs

```diff
@@ -2,8 +2,220 @@ mod operation;
+            .operations
+            .iter()
+            .any(|op| SKIPPED_OPERATIONS.contains(&op.name.as_str()))
+    });
+    if has_skipped_op {
+        return;
+    }
+
+    for test_case in test_file.tests {
+        if let Some(skip_reason) = test_case.skip_reason {
+            println!("Skipping {}: {}", test_case.description, skip_reason);
+            continue;
+        }
+
+        let client = create_client(test_case.client_options, test_case.use_multiple_mongoses).await;
+
+        if let Some(ref run_on) = test_file.run_on {
+            let can_run_on = run_on.iter().any(|run_on| run_on.can_run_on(&client));
+            if !can_run_on {
+                println!("Skipping {}", test_case.description);
+                continue;
+            }
+        }
+
+        let db_name = match test_file.database_name {
+            Some(ref db_name) => db_name.clone(),
+            None => test_case.description.replace('$', "%").replace(' ', "_"),
+        };
+
+        let coll_name = match test_file.collection_name {
+            Some(ref coll_name) => coll_name.clone(),
+            None => "coll".to_string(),
+        };
+
+        if test_case
+            .description
+            .contains("Aggregate with $listLocalSessions")
+        {
+            // TODO DRIVERS-1230: This test currently fails on 3.6 standalones because the session
+            // does not attach to the server ping. When the driver is updated to send implicit
+            // sessions to standalones, this test should be unskipped.
+            let req = semver::VersionReq::parse("<= 3.6").unwrap();
+            if req.matches(&client.server_version) && client.is_standalone() {
+                continue;
+            }
+            start_session(&client, &db_name).await;
+        }
+
+        if let Some(ref data) = test_file.data {
+            match data {
+                TestData::Single(data) => {
+                    if !data.is_empty() {
+                        let coll = if client.is_replica_set() || client.is_sharded() {
+                            let write_concern =
+                                WriteConcern::builder().w(Acknowledgment::Majority).build();
+                            let options = CollectionOptions::builder()
+                                .write_concern(write_concern)
+                                .build();
+                            client
+                                .init_db_and_coll_with_options(&db_name, &coll_name, options)
+                                .await
+                        } else {
+                            client.init_db_and_coll(&db_name, &coll_name).await
+                        };
+                        coll.insert_many(data.clone(), None)
+                            .await
+                            .expect(&test_case.description);
+                    }
+                }
+                TestData::Many(_) => panic!("{}: invalid data format", &test_case.description),
```

Comment:
why is this unimplemented?

Reply:
This is the data format used for gridfs tests -- since we don't have gridfs implemented, all those tests should skip which means reaching this branch would be an error in the test runner.

Reply:
Ah makes sense. I remember implementing this when doing retryable reads for Swift, but I guess I never verified whether it was necessary or not (we skip gridfs there too).